components/server/src/api/envvar-service-api.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { Code, ConnectError, HandlerContext, ServiceImpl } from "@connectrpc/connect";
+import { EnvironmentVariableService as EnvironmentVariableServiceInterface } from "@gitpod/public-api/lib/gitpod/v1/envvar_connect";
+import {
+    ListEnvironmentVariablesRequest,
+    ListEnvironmentVariablesResponse,
+    UpdateEnvironmentVariableRequest,
+    UpdateEnvironmentVariableResponse,
+    DeleteEnvironmentVariableRequest,
+    DeleteEnvironmentVariableResponse,
+    CreateEnvironmentVariableRequest,
+    CreateEnvironmentVariableResponse,
+} from "@gitpod/public-api/lib/gitpod/v1/envvar_pb";
+import { inject, injectable } from "inversify";
+import { EnvVarService } from "../user/env-var-service";
+import { PublicAPIConverter } from "@gitpod/gitpod-protocol/lib/public-api-converter";
+import { ProjectEnvVarWithValue, UserEnvVar, UserEnvVarValue } from "@gitpod/gitpod-protocol";
+
+@injectable()
+export class EnvironmentVariableServiceAPI implements ServiceImpl<typeof EnvironmentVariableServiceInterface> {
+    @inject(EnvVarService)
+    private readonly envVarService: EnvVarService;
+
+    @inject(PublicAPIConverter)
+    private readonly apiConverter: PublicAPIConverter;
+
+    async listEnvironmentVariables(
+        req: ListEnvironmentVariablesRequest,
+        context: HandlerContext,
+    ): Promise<ListEnvironmentVariablesResponse> {
+        // if (req.workspaceId) {
+        //     const envVars = await this.envVarService.listUserEnvVars(context.user.id, context.user.id);
+        // }
```

Comment:
Not sure about merging getWorkspaceEnvVars logic here too, the return value only has name and value, not id, is that fine? personally feels confusing to have many different return shapes for the same function